tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.cs

```diff
@@ -91,7 +91,7 @@ public async Task StartPollingAsync()
 
             while (!_cancellationSource.IsCancellationRequested)
             {
-                var isRcmEnabled = !string.IsNullOrEmpty(_discoveryService.ConfigurationEndpoint);
+                var isRcmEnabled = Volatile.Read(ref _isRcmEnabled);
                 var isProductRegistered = _products.Any();
 
                 if (isRcmEnabled && isProductRegistered)
```

Comment:
Should we also unsubscribe on shutdown? (code at the end of the class)

Reply:
_Currently_ that's not required because RCM is only created once, but I think it's a good idea, because we will need to change that soon!

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/3154/commits/860e6809f7004743bbc84a72f6b7b7e84e2ba3bd